tsl/test/expected/attach_chunk.out

```diff
@@ -0,0 +1,266 @@
+ERROR:  child table "incompatible_table" has different type for column "device"
+-- Try to attach a table with missing columns
+CREATE TABLE missing_col_table(id int, time timestamptz not null);
+CALL attach_chunk('attach_test', 'missing_col_table', '{"time": ["2025-07-01 05:00:00+3", "2025-07-07 05:00:00+3"], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  child table is missing column "device"
+-- Try to attach a table with extra columns
+CREATE TABLE extra_col_table(id int, time timestamptz not null, device int, temp float, extra_col int);
+CALL attach_chunk('attach_test', 'extra_col_table', '{"time": ["2025-07-01 05:00:00+3", "2025-07-07 05:00:00+3"], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  table "extra_col_table" contains column "extra_col" not found in parent "attach_test"
+-- Attach by non-owner is not allowed
+set role :ROLE_1;
+CALL attach_chunk('attach_test', 'regular_table_to_attach', '{"time": ["2025-07-01 05:00:00+3", "2025-07-07 05:00:00+3"], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  must be owner of table regular_table_to_attach
+set role :ROLE_DEFAULT_PERM_USER;
+CALL attach_chunk('attach_test', 'regular_table_to_attach', '{"time": ["2025-06-01 05:00:00+3", "2025-06-07 05:00:00+3"], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  chunk creation failed due to collision
+-- Attach hypertable as a chunk
+CALL attach_chunk('attach_test', 'hypertable_with_different_dimension', '{"time": ["2025-07-01 05:00:00+3", "2025-07-07 05:00:00+3"], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  cannot attach hypertable as a chunk
+\set ON_ERROR_STOP 1
+-- Test rollback behavior
+SELECT count(*) AS "PRE_ROLLBACK_CHUNKS" FROM _timescaledb_catalog.chunk WHERE hypertable_id = :'HYPERTABLE_ID'; \gset
+ PRE_ROLLBACK_CHUNKS 
+---------------------
+                   3
+(1 row)
+
+BEGIN;
+CREATE TABLE rollback_test_table(id int, time timestamptz not null, device int, temp float);
+ALTER TABLE rollback_test_table ADD CONSTRAINT attach_test_temp_check CHECK (temp > 0);
+CALL attach_chunk('attach_test', 'rollback_test_table', '{"time": ["2025-07-01 05:00:00+3", "2025-07-07 05:00:00+3"], "id": [-9223372036854775808, 1073741823]}');
+ROLLBACK;
+SELECT count(*) = :'PRE_ROLLBACK_CHUNKS' FROM _timescaledb_catalog.chunk WHERE hypertable_id = :'HYPERTABLE_ID';
+ ?column? 
+----------
+ t
+(1 row)
+
+CALL attach_chunk('attach_test', 'regular_table_to_attach', '{"time": ["2025-07-01 05:00:00", "2025-07-07 05:00:00"], "id": [-9223372036854775808, 1073741823]}');
+CALL detach_chunk('regular_table_to_attach');
+CALL attach_chunk('attach_test', 'regular_table_to_attach', '{"time": ["2025-07-01", "2025-07-07"], "id": [-9223372036854775808, 1073741823]}');
+CALL detach_chunk('regular_table_to_attach');
+CALL attach_chunk('attach_test', 'regular_table_to_attach', '{"time": ["Tue July 1 05:00:00 2024 GMT+3", "Mon July 7 05:00:00 2024 GMT+3"], "id": [-9223372036854775808, 1073741823]}');
```

Comment:
Looks like the test aborted prematurely here. You might want to do this in an ON_ERROR_STOP 0 block. Otherwise anything after this will not be executed due to aborting the test and it is not always easy to notice.